src/apps/lwaftr/lwaftr.lua

```diff
@@ -156,31 +164,23 @@ local function drop_ipv4(lwstate, pkt, pkt_src_link)
    return drop(pkt)
 end
 
-local transmit_icmpv6_reply, transmit_icmpv4_reply
-
-local function init_transmit_icmpv6_reply(lwstate)
-   assert(lwstate.icmpv6_rate_limiter_n_seconds > 0,
-      "Incorrect icmpv6_rate_limiter_n_seconds value, must be > 0")
-   assert(lwstate.icmpv6_rate_limiter_n_packets >= 0,
-      "Incorrect icmpv6_rate_limiter_n_packets value, must be >= 0")
-   local icmpv6_rate_limiter_n_seconds = lwstate.icmpv6_rate_limiter_n_seconds
-   local icmpv6_rate_limiter_n_packets = lwstate.icmpv6_rate_limiter_n_packets
+local function init_transmit_icmpv6_reply (rate_limiting)
    local num_packets = 0
    local last_time
-   return function (o, pkt)
+   return function (lwstate, pkt)
       local now = tonumber(engine.now())
       last_time = last_time or now
       -- Reset if elapsed time reached.
-      if now - last_time >= icmpv6_rate_limiter_n_seconds then
+      if now - last_time >= rate_limiting.period then
```

Comment:
According to my measurements indeed, no issue.  But it is for ICMP anyway so it's not on the hot hot path.